homeassistant/components/ituran/coordinator.py

```diff
@@ -0,0 +1,77 @@
+"""Coordinator for Ituran."""
+
+import logging
+
+from pyituran import Ituran, Vehicle
+from pyituran.exceptions import IturanApiError, IturanAuthError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_ID_OR_PASSPORT,
+    CONF_MOBILE_ID,
+    CONF_PHONE_NUMBER,
+    DOMAIN,
+    UPDATE_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+type IturanConfigEntry = ConfigEntry[IturanDataUpdateCoordinator]
+
+
+class IturanDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Vehicle]]):
+    """Class to manage fetching Ituran data."""
+
+    config_entry: IturanConfigEntry
+    ituran: Ituran
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize account-wide Ituran data updater."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN}-{entry.data[CONF_ID_OR_PASSPORT]}",
+            update_interval=UPDATE_INTERVAL,
+            config_entry=entry,
+        )
+        self.ituran = Ituran(
```

Comment:
actually, I just had an idea how to make this config flow a bit prettier (I never liked to optional field solution here tbh - sorry for coming up with this this late 😅): I'd suggest using a menu flow here like `habitica` does:
- I need a new OTP
- I have an existing OTP

that would make the flow a lot easier for the user imo

![image](https://github.com/user-attachments/assets/a1d90b1b-87cf-401a-aa9c-77142024374f)

Reply:
I understand your reasoning but the thing is that I don't want to put too much focus on the mobile ID since, out of the very few people that might end up using this integration, I'll probably be the only one to actually provide my own mobile ID.
When using Ituran, if you "log in" from one device, it'll invalidate the previous one since it only saves one mobile ID per account. This means that if I use my mobile phone and then add the integration to HA, the mobile phone will be logged out. My way around this, to make them both work simultaneously, was to sniff the network traffic from my phone (which is how I learned how to use their web service in the first place) and save the mobile ID my phone generated. I don't think many people will do that but I didn't want to force myself, or anyone else, to go and edit HA's internal files in case they did want to do something like or have multiple HA instances that work together, etc.

Reply:
i get that reasoning, let's wait on another opinion

Reply:
Mobile ID was removed